src/module.c

```diff
@@ -10784,6 +10906,18 @@ int moduleLoad(const char *path, void **module_argv, int module_argc) {
             "Module %s initialization failed. Module not loaded",path);
         return C_ERR;
     }
+    if (is_loadex && listLength(server.module_configs_queue)) {
+        moduleUnregisterCommands(ctx.module);
+        moduleUnregisterSharedAPI(ctx.module);
+        moduleUnregisterUsedAPI(ctx.module);
+        moduleRemoveConfigs(ctx.module);
+        moduleFreeModuleStructure(ctx.module);
+        moduleFreeContext(&ctx);
+        dlclose(handle);
+        serverLog(LL_WARNING,
+            "RedisModule_LoadConfigs not called with configs provided by loadex. Module %s not loaded.",path);
+        return C_ERR;
```

Comment:
this will fail if the user provided configs to LOADEX and they where either not recognized by the module, or it forgot to call the Load API, right?

maybe we should also cover the case that the module registered configs and forgot to call the Load API?
i.e. even if the user didn't pass configs, or used just MODULE LOAD and not MODULE LOADEX?
WDYT?